packages/blade/src/components/Alert/Alert.tsx

```diff
@@ -0,0 +1,217 @@
+import type { ReactChild, ReactElement } from 'react';
+import { useState } from 'react';
+
+import StyledAlert from './StyledAlert';
+import { AlertIcon, CloseIcon } from '~components/Icons';
+import { makeAccessible } from '~utils';
+import { IconButton } from '~components/Button/IconButton';
+import Box from '~components/Box';
+import { Heading, Text } from '~components/Typography';
+import BaseButton from '~components/Button/BaseButton';
+import { BaseLink } from '~components/Link/BaseLink';
+
+type Nullable<Type> = Type | null;
+
+type PrimaryAction = {
+  text: string;
+  onClick: () => void;
+};
+
+type SecondaryActionButton = {
+  text: string;
+  onClick: () => void;
+};
+
+type SecondaryActionLinkButton = {
+  text: string;
+  href: string;
+  onClick?: () => void;
+  target?: string;
+  /**
+   * When `target` is set to `_blank` this is automatically set to `noopener noreferrer`
+   */
+  rel?: string;
+};
+
+type SecondaryAction = SecondaryActionButton | SecondaryActionLinkButton;
+
+export type AlertProps = {
+  /**
+   * Body content, pass text or JSX. Avoid passing components except `Link` to customize the content.
+   */
+  description: ReactChild;
+
+  /**
+   * A brief heading
+   */
+  title?: string;
+
+  /**
+   * Shows a dismiss button
+   *
+   * @default true
+   */
+  isDismissable?: boolean;
+
+  /**
+   * A callback when the dismiss button is clicked
+   */
+  onDismiss?: () => void;
+
+  /**
+   * Can be set to `high` for a more prominent look. Not to be confused with a11y contrast.
+   *
+   * @default low
+   */
+  contrast?: 'high' | 'low';
+
+  /**
+   * Makes the Alert span the entire container width, instead of the default max width of `584px`
+   *
+   * @default false
+   */
+  isFullWidth?: boolean;
+
+  /**
+   * Sets the color tone
+   *
+   * @default information
+   */
+  intent?: 'positive' | 'negative' | 'information' | 'notice';
+
+  /**
+   * Removes border and border radii, useful for creating full bleed layouts. Automatically sets `isFullWidth` to `true` when enabled.
+   *
+   * @default false
+   */
+  isBorderless?: boolean;
+
+  /**
+   * Renders a primary action button and a secondary action link button
+   */
+  actions?: {
+    /**
+     * Renders a button
+     */
+    primary?: PrimaryAction;
+
+    /**
+     * Renders a Link button
+     */
+    secondary?: SecondaryAction;
+  };
+};
+
+// todo: add all icons
+const intentIconMap = {
+  positive: AlertIcon,
+  negative: AlertIcon,
+  information: AlertIcon,
+  notice: AlertIcon,
+};
+
+const Alert = ({
+  description,
+  title,
+  isDismissable = true,
+  onDismiss,
+  contrast = 'low',
+  isFullWidth = false,
+  intent = 'information',
+  isBorderless = false,
+  actions,
+}: AlertProps): ReactElement | null => {
+  const [isVisible, setIsVisible] = useState(true);
+  const contrastType = contrast === 'high' ? 'highContrast' : 'lowContrast';
```

Comment:
```suggestion
  const contrastType = `${contrast}Contrast`;
```

Reply:
Done in https://github.com/razorpay/blade/pull/654/commits/ee1c1bbada067d0954583da2d27b9ffd898dee65